File: Cart-backend/src/main/java/com/example/demo/service/CartService.java
package com.sliit.crave.service;


import com.sliit.crave.dto.response.Response;
import com.sliit.crave.dto.response.cart.CartResponse;
import com.sliit.crave.entity.Cart;

public interface CartService {
    Response<CartResponse> getCartByUserId(Long userId);
    Response<CartResponse> addItemToCart(Long userId, Long foodId, int quantity);
    Response<CartResponse> removeItemFromCart(Long userId, Long cartItemId);
    Response<String> clearCart(Long userId);
}

File: Cart-frontend/CART-FRONTEND/src/services/cart.service.js
import {api} from "../api.js";

export const getMyCart = async () => {
    try {
        const response = await api.get("/cart/me");
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const addItemToCart = async (foodId, quantity) => {
    try {
        const response = await api.post(`/cart/add`, null, { params: { foodId, quantity } });
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const removeItemFromCart = async (cartItemId) => {
    try {
        const response = await api.delete(`/cart/remove/${cartItemId}`);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};

export const clearCart = async () => {
    try {
        const response = await api.delete(`/cart/clear`);
        return response.data;
    } catch (e) {
        console.error(e);
    }
};


